runtime/src/lib.rs

```diff
@@ -146,17 +150,54 @@ impl Drivers {
         let hashed_rt_pub_key = crypto
             .hash(AlgLen::Bit384, &rt_pub_key.to_der()[1..])
             .map_err(|_| CaliptraError::RUNTIME_INITIALIZE_DPE_FAILED)?;
+        let mut cert = [0u8; MAX_CERT_CHAIN_SIZE];
+        let data_vault = DataVault::new(DvReg::new());
+        let ldevid_cert_size =
+            dice::copy_ldevid_cert(&data_vault, persistent_data.get(), &mut cert)
+                .map_err(|_| CaliptraError::RUNTIME_CERT_CHAIN_CREATION_FAILED)?;
+        if ldevid_cert_size > cert.len() {
+            return Err(CaliptraError::RUNTIME_CERT_CHAIN_CREATION_FAILED);
+        }
+        let fmcalias_cert_size = dice::copy_fmc_alias_cert(
+            &data_vault,
+            persistent_data.get(),
+            &mut cert[ldevid_cert_size..],
+        )
+        .map_err(|_| CaliptraError::RUNTIME_CERT_CHAIN_CREATION_FAILED)?;
+        if ldevid_cert_size + fmcalias_cert_size > cert.len() {
+            return Err(CaliptraError::RUNTIME_CERT_CHAIN_CREATION_FAILED);
+        }
+        let rtalias_cert_size = dice::copy_rt_alias_cert(
+            &data_vault,
+            persistent_data.get(),
+            &mut cert[ldevid_cert_size + fmcalias_cert_size..],
+        )
+        .map_err(|_| CaliptraError::RUNTIME_CERT_CHAIN_CREATION_FAILED)?;
+        let cert_chain_size = ldevid_cert_size + fmcalias_cert_size + rtalias_cert_size;
+        if cert_chain_size > cert.len() {
+            return Err(CaliptraError::RUNTIME_CERT_CHAIN_CREATION_FAILED);
+        }
+        let mut cert_chain = ArrayVec::<u8, MAX_CERT_CHAIN_SIZE>::new();
+        for i in 0..cert_chain_size {
+            cert_chain
+                .try_push(
+                    *cert
+                        .get(i)
+                        .ok_or(CaliptraError::RUNTIME_CERT_CHAIN_CREATION_FAILED)?,
+                )
+                .map_err(|_| CaliptraError::RUNTIME_CERT_CHAIN_CREATION_FAILED)?;
+        }
```

Comment:
Can you move the cert chain building to a helper function? I think that will make the driver init code easier to understnand.

Reply:
Agree, done